src/coreclr/jit/fgopt.cpp

```diff
@@ -5345,7 +5232,8 @@ bool Compiler::ThreeOptLayout::CompactHotJumps()
             // If we aren't sure which successor is hotter, and we already fall into one of them,
             // do nothing.
             BasicBlock* const unlikelyTarget = unlikelyEdge->getDestinationBlock();
-            if ((unlikelyEdge->getLikelihood() == 0.5) && (unlikelyTarget->bbPostorderNum == (i + 1)))
+            if ((unlikelyEdge->getLikelihood() == 0.5) && isCandidateBlock(unlikelyTarget) &&
+                (unlikelyTarget->bbPreorderNum == (i + 1)))
```

Comment:
I forgot to check if the unlikely target is in the candidate span of blocks before checking its ordinal. This isn't a correctness issue per se, but switching ordinals from `bbPostorderNum` to `bbPreorderNum` created churn here.